encode/vp9enc.c

```diff
@@ -814,8 +819,62 @@ vp9enc_update_reference_list()
    }
 }
 
+static
+void vp9enc_set_refreshparameter_for_svct_2layers(VAEncPictureParameterBufferVP9 *pic_param, int current_frame, int *is_golden_refreshed)
+{
+  //Pattern taken from libyami
+
+  pic_param->ref_flags.bits.ref_frame_ctrl_l0 = 1;
+  if(*is_golden_refreshed && (current_frame % 2) != 0)
+    pic_param->ref_flags.bits.ref_frame_ctrl_l0 |= 0x2;
+
+
+  switch(current_frame % 2) {
+    case 0:
+      //Layer 0
+      pic_param->refresh_frame_flags = 0x01;
+      break;
+    case 1:
+      //Layer 1
+      pic_param->refresh_frame_flags = 0x02;
+      *is_golden_refreshed = 1;
+      break;
+  }
+
+}
+
+static
+void vp9enc_set_refreshparameter_for_svct_3layers(VAEncPictureParameterBufferVP9 *pic_param, int current_frame,int *is_golden_refreshed)
+{
+  //Pattern taken from libyami - Note that the alternate frame is never referenced,
+  //this is because, libyami implementation suggests to be able to drop individual
+  //frames from Layer 2 on bad network connections
+
+  pic_param->ref_flags.bits.ref_frame_ctrl_l0 = 0x1;
+  if(*is_golden_refreshed && (current_frame % 4) != 0)
+    pic_param->ref_flags.bits.ref_frame_ctrl_l0 |= 0x2;
+
+  switch(current_frame % 4) {
+    case 0:
+      //Layer 0
+      pic_param->refresh_frame_flags = 0x01;
+      break;
+    case 1:
+    case 3:
+      //Layer 2
+      pic_param->refresh_frame_flags = 0x00;
+      break;
+    case 2:
+      //Layer 1
+      pic_param->refresh_frame_flags = 0x02;
+      *is_golden_refreshed = 1;
+      break;
+  }
+
+}
+
```

Comment:
Both functions are quite similar, perhaps their merge worth, with a parameter with the number of layers

Reply:
thats true - but for vp8enc I also used 2 functions (for 2 and 3 Layers), which is already merged. So I'd like to keep it this way for consistency reasons